### ISSUE TITLE
Prevent Dymo Connect opening when plug-in a printer USB

### ISSUE THREAD

**Issue by ZZJHONS** [dymosoftware/dymo-connect-framework#14]
Is there an option on Dymo Connect Software to prevent its opening each time a USB from a printer is plugged to the PC?

We use our software to generate PDF labels for a Dymo 4XL and a 450, the Dymo service is always running by default, there is no need to open Dymo Connect each time we plug a printer to the PC.

Is there a way to disable it? Or can it be implemented?

Thanks

**Comment by Bea5tOfB0urb0n** [comment]
Having the same problem here. Unbelieveably annoying. I have the printer plugged into my docking station so every morning when I log in, Dymo Connect is there right in the front. I want this to stop. The fact this was even written into the application is gobsmackingly insane, was the dev team completely high when they worked on that bit?

And this first comment was 6 months ago. Great support Dymo, good to see you actually care about your customers.

**Comment by d-bugger** [comment]
Annoying indeed ... there should be an option in the preferences to disable this!

**Comment by thijsvanbrabant** [comment]
Each time I'm logged out and I sign in again. There it is...:-S Even when the screen just locks.
I also have it connected to my USB dockingstation.

**Comment by HighMans** [comment]
Found a solution to this POS software.

First kill all applications that start with DYMO (dymoconnect.exe, dymoconnectlauncher.exe, etc.) using task manager.

In C:\Program Files (x86)\Dymo\Dymo Connect or C:\Program Files\Dymo\Dymo Connect rename the file called "DymoConnectLauncher.exe" to "DymoConnectLauncher.exe.crappysoftware".

Reboot your Dymo and the connect software should no longer launch upon plugging in.

**Comment by SBCV-apegram** [comment]
> Found a solution to this POS software.
> 
> First kill all applications that start with DYMO (dymoconnect.exe, dymoconnectlauncher.exe, etc.) using task manager.
> 
> In C:\Program Files (x86)\Dymo\Dymo Connect or C:\Program Files\Dymo\Dymo Connect rename the file called "DymoConnectLauncher.exe" to "DymoConnectLauncher.exe.crappysoftware".
> 
> Reboot your Dymo and the connect software should no longer launch upon plugging in.

Thanks so much for this! I'm going one step further and pushing this out as GPO to remove the file and disable the DYMOConnectPnP service. My end users are going to be thrilled (we're all on docking stations too)...

**Comment by ProNotion** [comment]
Disappointing to see this has still not been addressed. It's quite intrusive and extremely annoying not having control over this configuration.

**Comment by tspaeth** [comment]
Just to bring in another device class to this annoyance:
I've got a PC with a second monitor and a USB hub attached to it. As soon as I power up the monitors in the morning, the USB hub is "activated" and thus the dymo software popups up every day. 
This is even worse than a printer - powerup...

**Comment by ARGO1960** [comment]
Yes annoying is is this phone home we will always show full screen when we start software. 
But to disable this you can do that easier without renaming and searching.
run: MSCONFIG.exe
choose the services TAB:
scroll down to you find: DYMO connect PNP service. and disableit.
Save all.
you don't have to reboot the next time you do the app wont start anymore so you need to start the DYMO software yourself
![disable_dymo](https://user-images.githubusercontent.com/14184860/217184915-d2b74846-a611-4a78-842f-bb80762259ee.png)

**Comment by ProNotion** [comment]
@ARGO1960 Unfortunately this didn't work for me on Windows 11 Pro

**Comment by ARGO1960** [comment]
> @ARGO1960 Unfortunately this didn't work for me on Windows 11 Pro

Well yes i'm using 11 Enterprise but msconfig should work on all windows versions as I recall.
what can't be done in your case? does msconfig not start. 
if you reconnect the printer while not rebooted it will start the app again as the service is still running.

**Comment by ProNotion** [comment]
> > @ARGO1960 Unfortunately this didn't work for me on Windows 11 Pro
> 
> Well yes i'm using 11 Enterprise but msconfig should work on all windows versions as I recall. what can't be done in your case? does msconfig not start. if you reconnect the printer while not rebooted it will start the app again as the service is still running.

I have no problem opening MsConfig, I meant disabling the DYMO Connect PnP Service and rebooting afterwards had no effect. As soon as I plugged the printer back in after the reboot the app opened up again. I double-checked the service was still disabled after the reboot and it was.

**Comment by ARGO1960** [comment]
> > > @ARGO1960 Unfortunately this didn't work for me on Windows 11 Pro
> > 
> > 
> > Well yes i'm using 11 Enterprise but msconfig should work on all windows versions as I recall. what can't be done in your case? does msconfig not start. if you reconnect the printer while not rebooted it will start the app again as the service is still running.
> 
> I have no problem opening MsConfig, I meant disabling the DYMO Connect PnP Service and rebooting afterwards had no effect. As soon as I plugged the printer back in after the reboot the app opened up again. I double-checked the service was still disabled after the reboot and it was.

You know what your absolutly right!, I checked and yes it still does. :-( (don't get why it did not the first time)

But then we look in a different location: Start task manager -> then select the "Speed tab" and deiable the startups there.
now i tried 3 times this should work.
![disable-dymo2](https://user-images.githubusercontent.com/14184860/217237779-20fa6713-94ee-4bf1-a7fd-ea51d743a30a.png)
I disabled all from Dymo but i suspect only "Dymoconnectlauncher" is the one you need to disable.

**Comment by konradatwork** [comment]
You can also disable the Service from automatically running on login.

**Comment by ThaDaVos** [comment]
For anyone running into this issue on Windows Server or a like - go to "Settings" > "Apps" > "Startup" and disable it there
(This is because on Windows Server there's no Startup tab in the Task Manager)

**Comment by J-Setzer** [comment]
Really annoying! Popup up even **every** time I reactivate Win11 from energy save mode.

**Comment by ProNotion** [comment]
I no longer have the issue. I believe it is because of the previous advice on disabling startup apps.  My Startup apps are now disabled like this.

![image](https://github.com/dymosoftware/dymo-connect-framework/assets/1590398/99681537-90a3-42a8-a71e-e0509e6ec839)

**Comment by J-Setzer** [comment]
I've disabled service, stopped process and removed from autostart months ago.
But it started again annoying me after I have used the App once, yesterday.

**Comment by gdgib** [comment]
FWIW, if you're on a system where msconfig doesn't show startup options, you can manually remove this from `HKCU/SOFTWARE/Microsoft/Windows/CurrentVersion/Run`.  I suppose on some systems in might be under HKLM instead of HKCU.

Obviously please do be careful editing the registry of course, and consider backing up the relevant key before deletion (as if anyone would ever want this "feature" restored).

**Comment by StarCrusherDust** [comment]
How about the security vulnerability this creates for end users to exploit?  In my testing immediately after the software is deployed (if deployed via managed platform) the dymoconnectlauncher.exe is left running in an elevated context and on printer connection it launches the dymo connect software in an elevated context as well.  Now all a semi-knowledgeable user needs to do is open a file dialog window such as file open, navigate to windows system32 and and *.exe and press enter then browse and run cmd.exe, now they can do anything to the system they want…

**Comment by SBCV-apegram** [comment]
> How about the security vulnerability this creates for end users to exploit? In my testing immediately after the software is deployed (if deployed via managed platform) the dymoconnectlauncher.exe is left running in an elevated context and on printer connection it launches the dymo connect software in an elevated context as well. Now all a semi-knowledgeable user needs to do is open a file dialog window such as file open, navigate to windows system32 and type *.exe and press enter then browse and run cmd.exe, now they can do anything to the system they want…

I remember using that exploit in middle school 20 years... I thought Microsoft would have closed that loophole years ago, and yet it still works perfectly (I just checked). Until this vendor gets their act together, be sure to add a mitigation action to the end of the deployment script, such as a reboot or a force kill of the dymoconnectlauncher process. 🙄

**Comment by SBCV-apegram** [comment]
@dymosoftware does anyone check these issues? This has been an open request since 2021, and now a security vulnerability has been discovered. Please do something!

**Comment by xioTechnologies** [comment]
No one has reported if just disabling the start up (image below) solves the problem or not.  I think this solved the issue on my old machine but I'm yet to see if it works with latest versions.

<img width="768" alt="image" src="https://github.com/user-attachments/assets/9103fc21-6a95-478f-bcf8-e67044f3aedd">

**Comment by michaelstec** [comment]
I will try uninstalling the DYMO Connect Web Service portion of the app and report back
